docs/messaging/messaging-spans.md

```diff
@@ -153,38 +153,32 @@ in such a way that it cannot be changed by intermediaries.
 
 ### Span name
 
-The span name SHOULD be set to the message destination name and the name of the operation being performed (as captured in [`messaging.operation.name`](../attributes-registry/messaging.md)) in the following format:
+Messaging spans MUST follow the overall [guidelines for span names](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/trace/api.md#span).
 
-```
-<destination name> <operation name>
-```
+<!-- markdown-link-check-disable -->
+<!-- HTML anchors are not supported https://github.com/tcort/markdown-link-check/issues/225-->
+The **span name** SHOULD be `{messaging.operation.name} {destination}` (see below for the exact definition of the [`{destination}`](#destination-placeholder) placeholder).
+<!-- markdown-link-check-enable -->
 
-If the operation name is not specified by the messaging system, then the operation type as defined in [Operation types](#operation-types) SHOULD be used:
+Semantic conventions for individual messaging systems MAY specify different span name format and then MUST document it in [semantic conventions for specific messaging technologies](#semantic-conventions-for-specific-messaging-technologies).
 
-```
-<destination name> <operation type>
-```
+The <span id="destination-placeholder">`{destination}`</span> SHOULD adhere to one of the following values, arranged in prioritized order, provided they are accessible:
 
-The destination name SHOULD only be used for the span name if it is known to be of low cardinality (cf. [general span name guidelines](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/trace/api.md#span)).
-This can be assumed if it is statically derived from application code or configuration.
-Wherever possible, the real destination names after resolving logical or aliased names SHOULD be used.
-If the destination name is dynamic, such as a [conversation ID](#conversations) or a value obtained from a `Reply-To` header, it SHOULD NOT be used for the span name.
-In these cases, an artificial destination name that best expresses the destination, or a generic, static fallback like `"(anonymous)"` for [anonymous destinations](#temporary-and-anonymous-destinations) SHOULD be used instead.
+- `messaging.destination.template`
+- `messaging.destination.name` if it is known to be of a low cardinality.
```

Comment:
can we say anything more specific than "low cardinality"? my specific concern is that generic messaging instrumentation may not be able to guarantee someone doesn't have 10,000 destinations and so the generic instrumentation might choose not to put this in the span name which I don't think is what we want

Reply:
```suggestion
- `messaging.destination.name` if it is statically derived from application code or configuration.
```

Maybe that's better? This will result in a low cardinality in most cases, and avoids that instrumentation authors need to reason about what high or low cardinality means.

Reply:
I rephrased to 

> The `messaging.destination.template` SHOULD be used whenever it's available. Otherwise, `messaging.destination.name` SHOULD be used only when
  the destination is known to be neither [temporary nor anonymous](#temporary-and-anonymous-destinations).

WDYT?

(also aligned with our DB span name changes in #1069)